pinot-spi/src/main/java/org/apache/pinot/spi/filesystem/PinotFS.java

```diff
@@ -70,6 +70,16 @@ boolean mkdir(URI uri)
   boolean delete(URI segmentUri, boolean forceDelete)
       throws IOException;
 
+  /**
+   * Deletes the files at the locations provided. If the segmentUri is a directory, it will delete the entire directory.
+   * @param segmentUris List of URIs of the segments
+   * @param forceDelete true if we want the uri and any sub-uris to always be deleted, false if we want delete to fail
+   * @return true if delete is successful else false
```

Comment:
What if the delete only partially succeeded? Not all filesystems support batch deletion so some implementation will have to issue the delete in a loop.